partner_multi_company/tests/test_partner_multi_company.py

```diff
@@ -0,0 +1,175 @@
+            self.company_1.id,
+        )
+        partner = self.env["res.partner"].create(
+            {"name": "Test 2", "company_ids": [(5, False)]}
+        )
+        self.assertFalse(partner.company_id)
+
+    def test_company_none(self):
+        self.assertFalse(self.partner_company_none.company_id)
+        # All of this should be allowed
+        self.partner_company_none.with_user(self.user_company_1.id).name = "Test"
+        self.partner_company_none.with_user(self.user_company_2.id).name = "Test"
+
+    def test_company_1(self):
+        self.assertEqual(self.partner_company_1.company_id, self.company_1)
+        # All of this should be allowed
+        self.partner_company_1.with_user(self.user_company_1).name = "Test"
+        self.partner_company_both.with_user(self.user_company_1).name = "Test"
+        # And this one not
+        with self.assertRaises(AccessError):
+            self.partner_company_2.with_user(self.user_company_1).name = "Test"
+
+    def test_create_company_1(self):
+        partner = self.partner_model.with_user(self.user_company_1).create(
+            {
+                "name": "Test from user company 1",
+                "company_ids": [(6, 0, self.company_1.ids)],
+            }
+        )
+        self.assertEqual(partner.company_id, self.company_1)
+
+    def test_create_company_2(self):
+        partner = self.partner_model.with_user(self.user_company_2).create(
+            {
+                "name": "Test from user company 2",
+                "company_ids": [(6, 0, self.company_2.ids)],
+            }
+        )
+        self.assertEqual(partner.company_id, self.company_2)
+
+    def test_company_2(self):
+        self.assertEqual(self.partner_company_2.company_id, self.company_2)
+        # All of this should be allowed
+        self.partner_company_2.with_user(self.user_company_2).name = "Test"
+        self.partner_company_both.with_user(self.user_company_2).name = "Test"
+        # And this one not
+        with self.assertRaises(AccessError):
+            self.partner_company_1.with_user(self.user_company_2).name = "Test"
+
+    def test_uninstall(self):
+        from ..hooks import uninstall_hook
+
+        uninstall_hook(self.env.cr, None)
+        rule = self.env.ref("base.res_partner_rule")
+        domain = (
+            "['|','|',"
+            "('company_id.child_ids','child_of',[user.company_id.id]),"
+            "('company_id','child_of',[user.company_id.id]),"
+            "('company_id','=',False)]"
+        )
+        self.assertEqual(rule.domain_force, domain)
+        self.assertFalse(rule.active)
+
+    def test_switch_user_company(self):
+        self.user_company_1.company_ids = (self.company_1 + self.company_2).ids
+        self.user_company_1.company_id = self.company_2.id
+        self.user_company_1 = self.user_company_1.with_user(self.user_company_2)
+        self.assertEqual(
+            self.user_company_1.partner_id.company_id,
```

Comment:
When you change user's company it should not change related partner's company. I think here should be just self.user_company_1.company_id

Reply:
@JasminSForgeFlow can you check this? :)